diff --git a/main.py b/main.py
index 21adf31..31cdd28 100644
--- a/main.py
+++ b/main.py
@@ -56,7 +56,7 @@ class MatrixClient(nio.AsyncClient):
         await self.sync(timeout)

         # Set up event callbacks after syncing once to ignore old messages.
-        callbacks = Callbacks(self)
+        callbacks = Callbacks(self, discord_client)

         self.add_event_callback(
             callbacks.message_callback,
@@ -351,8 +351,9 @@ class DiscordClient(discord.ext.commands.Bot):


 class Callbacks(object):
-    def __init__(self, matrix_client):
+    def __init__(self, matrix_client, discord_client):
         self.matrix_client = matrix_client
+        self.discord_client = discord_client

     def get_channel(self, room):
         channel_id = next(
@@ -500,7 +501,9 @@ class Callbacks(object):
             # :emote: becomes <:emote:emote_id>
             if emote not in added_emotes:
                 added_emotes.append(emote)
-                emote_ = discord.utils.get(guild.emojis, name=emote)
+                emote_ = discord.utils.get(
+                    self.discord_client.emojis, name=emote
+                )
                 if emote_:
                     message = message.replace(f":{emote}:", str(emote_))
